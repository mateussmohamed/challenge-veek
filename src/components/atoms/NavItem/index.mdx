---
name: NavItem
menu: Atoms
---

import { Playground, PropsTable } from 'docz'
import NavItem from './index'

# NavItem
<PropsTable of={NavItem} />

<Playground>
  <NavItem>I am link</NavItem>
</Playground>
