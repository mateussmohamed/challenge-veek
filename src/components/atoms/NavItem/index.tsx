import styled from 'styled-components'

import { Link } from '../Link'

export const NavItem = styled(Link)`
  color: #000;
  :nth-of-type(n + 5) {
    order: 2;
  }

  @media only screen and (min-width: 0em) {
    font-size: 16px;
  }

  @media only screen and (min-width: 48em) {
    margin: 0;
    padding: 0;
    text-align: normal;
  }
`
