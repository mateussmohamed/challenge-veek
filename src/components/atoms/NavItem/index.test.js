import React from 'react'
import { shallow } from 'enzyme'
import 'jest-styled-components'

import NavItem from './index'

describe('[<NavItem />]', () => {
  const wrap = (props = {}) => shallow(<NavItem {...props} />)

  test('matches to snapshot', () => {
    const wrapper = wrap({ children: 'I am a NavItem' })

    expect(wrapper).toMatchSnapshot()
  })
})
