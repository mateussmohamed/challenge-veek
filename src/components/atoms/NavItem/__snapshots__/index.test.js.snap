// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[<NavItem />] matches to snapshot 1`] = `
<StyledComponent
  forwardedClass={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "NavItem-e5pex5-0",
        "isStatic": true,
        "rules": Array [
          "color:#000;:nth-of-type(n + 5){order:2;}@media only screen and (min-width:0em){font-size:16px;}@media only screen and (min-width:48em){margin:0;padding:0;text-align:normal;}",
        ],
      },
      "displayName": "NavItem",
      "render": [Function],
      "styledComponentId": "NavItem-e5pex5-0",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
      Symbol(Symbol.hasInstance): [Function],
    }
  }
  forwardedRef={null}
>
  I am a NavItem
</StyledComponent>
`;
